community/kernel/src/test/java/org/neo4j/kernel/impl/storemigration/StoreVersionCheckTest.java

```diff
@@ -89,11 +92,29 @@ public void shouldReportFileWithCorrectVersion() throws IOException
         assertTrue( storeVersionCheck.hasVersion( shortFile, "correctVersion 1" ).outcome.isSuccessful() );
     }
 
-    private File fileContaining( EphemeralFileSystemAbstraction fs, String content ) throws IOException
+    @Test
+    public void shouldReportFileWithCorrectVersionWhenTrailerSpansMultiplePages() throws IOException
+    {
+        // given
+        FileSystemAbstraction fs = new DefaultFileSystemAbstraction();
+        PageCache pageCache = pageCacheRule.getPageCache( fs );
+        String trailer = "correctVersion 1";
+        byte[] trailerData = UTF8.encode( trailer );
+        byte[] data = new byte[pageCache.pageSize() + trailerData.length / 2];
+        System.arraycopy( trailerData, 0, data, pageCache.pageSize() - trailerData.length / 2, trailerData.length );
+        File shortFile = fileContaining( fs, UTF8.decode( data ) );
+
+        StoreVersionCheck storeVersionCheck = new StoreVersionCheck( pageCache );
+
+        // then
+        assertTrue( storeVersionCheck.hasVersion( shortFile, trailer ).outcome.isSuccessful() );
+    }
+
+    private File fileContaining( FileSystemAbstraction fs, String content ) throws IOException
     {
-        File shortFile = new File( "shortFile" );
+        File shortFile = File.createTempFile( "shortFile", "" );
```

Comment:
You could add a `deleteOnExit` call to this file, to get it cleaned up more eagerly.